src/applications/disability-benefits/all-claims/config/submitForm.js

```diff
@@ -37,31 +40,43 @@ const submitFormFor = eventName =>
             error = new Error(`vets_server_error: ${req.statusText}`);
           }
           error.statusText = req.statusText;
+          clearTimeout(timer);
           reject(error);
         }
       });
 
       req.addEventListener('error', () => {
         const error = new Error('client_error: Network request failed');
         error.statusText = req.statusText;
+        clearTimeout(timer);
         reject(error);
       });
 
       req.addEventListener('abort', () => {
         const error = new Error('client_error: Request aborted');
         error.statusText = req.statusText;
+        clearTimeout(timer);
         reject(error);
       });
 
       req.addEventListener('timeout', () => {
         const error = new Error('client_error: Request timed out');
         error.statusText = req.statusText;
+        clearTimeout(timer);
         reject(error);
       });
 
       req.setRequestHeader('X-Key-Inflection', 'camel');
       req.setRequestHeader('Content-Type', 'application/json');
 
+      // Throw an error after 30 seconds
+      timer = setTimeout(() => {
+        const error = new Error('client_error: Request taking too long');
+        // eslint-disable-next-line no-console
+        console.log(req, form, JSON.parse(body));
+        reject(error);
+      }, 3e4);
```

Comment:
Fancy `3e4` 😝 

We could also capture this in Sentry and see if we can find a case off of screenshare - [example](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/static-pages/renderHomepageBanner.js#L83)

Reply:
I think you could do both - log it the way you're doing already and also capture some details to Sentry

Reply:
Updated, please take a look. I'm not sure how much PII is appropriate for Sentry, so I commented those lines out.

Reply:
And I can't really test this right now since id.me is down.

Reply:
The Sentry part LGTM! I'm not sure about PII in Sentry either, maybe VSP will know.

The Google Analytic part, `recordEvent`, is up to you. Some teams do like to use GA to see how often an error is happening.